src/sentry/incidents/logic.py

```diff
@@ -1002,6 +995,24 @@ def update_alert_rule_trigger_action(
     return trigger_action
 
 
+def get_alert_rule_trigger_action_slack_channel_id(organization, integration_id, name):
+    from sentry.integrations.slack.utils import get_channel_id
+
+    _prefix, channel_id, timed_out = get_channel_id(organization, integration_id, name,)
+    if timed_out:
+        raise InvalidTriggerActionError(
+            "Could not find channel %s. We have timed out trying to look for it. " % name
```

Comment:
This is probs for a future PR but we should be implementing the async channel_id look up for metric alerts. That's why we have a hardcoded time_to_quit.